feat(docs): modify references pr request to add a comment + add references to node docs

sklppy88 opened the pull request:
This modifies the references frontmatter PR request to add a comment describing the files changed and what docs reference them.

critesjosh commented on docs/scripts/check_doc_references.sh:
```diff
@@ -136,13 +143,17 @@ REFERENCE_FILES=$(
-)
+  ' "$doc_file"
+done > "$MAPPING_FILE"
+
```
What do you think about adding a check that ensures all of the files that are referenced in the docs front matter actually exist? I think it would help us keep the docs in `references` clean and help us catch any misspellings.

Something like this:
```suggestion
# Validate all referenced files exist
echo "Validating referenced files exist..."
MISSING_FILES=""
while IFS='|' read -r ref_file doc_file; do
  if [[ ! -f "$ref_file" ]]; then
    MISSING_FILES="${MISSING_FILES}  - ${ref_file} (referenced in ${doc_file})\n"
  fi
done < "$MAPPING_FILE"

if [[ -n "$MISSING_FILES" ]]; then
  echo ""
  echo "ERROR: The following referenced files do not exist:"
  echo -e "$MISSING_FILES"
  echo "Please update the 'references' frontmatter in the affected documentation files."
  exit 1
fi
```

critesjosh commented on docs/scripts/check_doc_references.sh:
```diff
@@ -213,26 +238,88 @@ echo ""
+  IFS='|' read -ra DOC_ARRAY <<< "$docs"
+  for doc_file in "${DOC_ARRAY[@]}"; do
+    # Extract title from markdown file
+    DOC_TITLE=$(grep -m 1 "^title:" "$doc_file" 2>/dev/null | sed 's/^title: *//; s/^["'\'']*//; s/["'\'']*$//' || basename "$doc_file" .md)
```
DOC_TTILE and DOC_PATH arent used. should it be removed?

sklppy88 committed:
feat(docs): modify references pr request to add a comment + add references to node docs

This modifies the references frontmatter PR request to add a comment describing the files changed and what docs reference them.
